config/build/remove-test-types-from-dist.ts

```diff
@@ -0,0 +1,38 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as fglob from "fast-glob";
+
+const rootDir = path.join(__dirname, "..");
+const testFiles = fglob.sync("packages/*/dist/**/*.test.d.ts", {
+    cwd: rootDir,
+});
+
+for (const file of testFiles) {
+    fs.unlinkSync(path.join(rootDir, file));
+}
+
+const filesInTestDirs = fglob.sync("packages/*/dist/**/__tests__/*.d.ts", {
+    cwd: rootDir,
+});
+
+for (const file of filesInTestDirs) {
+    fs.unlinkSync(path.join(rootDir, file));
+}
+
+const dirs = fglob.sync("packages/*/dist/**/__tests__", {
+    cwd: rootDir,
+    onlyFiles: false,
+});
+
+for (const dir of dirs) {
+    fs.rmdirSync(path.join(rootDir, dir));
+}
+
+const filesInUtils = fglob.sync("utils/**/*.d.ts", {
+    cwd: rootDir,
+});
+
+for (const file of filesInUtils) {
+    fs.unlinkSync(path.join(rootDir, file));
+}
```

Comment:
I realize this is just an infra file, but I think it'd be much clearer to read if we extracted a helper function that accepted the globs and an option saying if we're deleting files or directories.  Then the rest of the file could read a bit more declaratively about what we're cleaning up... something like this: 

```suggestion
const deletePaths = (pathGlob, globOptions) => {
    const entries = fglob.sync(pathGlob, {
        cwd: rootDir,
       ...globOptions,
    });

    for (const entry of entries) {
        fs.rm(path.join(rootDir, entry), {recursive: true, force: true});
    }
}

deletePaths("packages/*/dist/**/*.test.d.ts")
deletePaths("packages/*/dist/**/__tests__/*.d.ts")
deletePaths("packages/*/dist/**/__tests__",{onlyDirectories: true})
deletePaths("utils/**/*.d.ts")
```

Reply:
I like this suggestion, sorry I didn't get to use it.